local_planner/src/nodes/planner_functions.cpp

```diff
@@ -439,23 +439,41 @@ float costFunction(float e_angle, float z_angle, float obstacle_distance,
 bool getDirectionFromTree(
     PolarPoint& p_pol,
     const std::vector<geometry_msgs::Point>& path_node_positions,
-    const Eigen::Vector3f& position) {
+    const Eigen::Vector3f& position, const Eigen::Vector3f& goal) {
   int size = path_node_positions.size();
   bool tree_available = true;
 
-  if (size > 0) {
+  if (size >
+      1) {  // path contains at least 2 points (current position and one wp)
+
+    // extend path with a node at the end in goal direction (for smoother
+    // transition to direct flight)
+    float node_distance =
+        (toEigen(path_node_positions[0]) - toEigen(path_node_positions[1]))
+            .norm();
+    Eigen::Vector3f dir_last_node_to_goal =
+        (goal - toEigen(path_node_positions[0])).normalized();
+    geometry_msgs::Point goal_node =
+        toPoint(toEigen(path_node_positions[0]) +
+                node_distance * dir_last_node_to_goal);
+    std::vector<geometry_msgs::Point> path_node_positions_extended;
+    path_node_positions_extended.push_back(goal_node);
+    path_node_positions_extended.insert(path_node_positions_extended.end(),
+                                        path_node_positions.begin(),
+                                        path_node_positions.end());
+    int size_extended = path_node_positions_extended.size();
+
+    // find path nodes between which the drone is currently located
     int min_dist_idx = 0;
     int second_min_dist_idx = 0;
     float min_dist = HUGE_VAL;
     float second_min_dist = HUGE_VAL;
-    float node_distance =
-        (toEigen(path_node_positions[0]) - toEigen(path_node_positions[1]))
-            .norm();
-
     std::vector<float> distances;
-    distances.reserve(size);
-    for (int i = 0; i < size; i++) {
-      distances.push_back((position - toEigen(path_node_positions[i])).norm());
+    distances.reserve(size_extended);
```

Comment:
I think you can construct it directly with the proper size: `std::vector<float> distances(size_extended);`